File: hidden-api/src/main/java/android/view/DisplayHidden.java
package android.view;

import android.os.Build;

import dev.rikka.tools.refine.RefineAs;

@RefineAs(Display.class)
public class DisplayHidden {
    public int getType() { throw new RuntimeException("Stub!"); }
    public int getLayerStack() { throw new RuntimeException("Stub!"); }
    public boolean getDisplayInfo(DisplayInfo outDisplayInfo) { throw new RuntimeException("Stub!"); }
}

File: hidden-api-contract/src/main/java/com/xaxaxax/moonclicker/hiddenapi/contracts/DisplayHidden.kt
package com.xaxaxax.moonclicker.hiddenapi.contracts

internal val DISPLAY_HIDDEN = StubContracts(
    stubs = mapOf("android.view.DisplayHidden" to "android.view.Display"),
    contracts = listOf(
        MemberContract(
            owner = "android.view.Display",
            member = MethodMember(name = "getType", returns = "int"),
        ),
        MemberContract(
            owner = "android.view.Display",
            member = MethodMember(name = "getLayerStack", returns = "int"),
        ),
        MemberContract(
            owner = "android.view.Display",
            member = MethodMember(
                name = "getDisplayInfo",
                parameters = listOf("android.view.DisplayInfo"),
                returns = "boolean",
            ),
        ),
    ),
)
